### ISSUE TITLE
Support for pyproject.toml dependencies (PEP518)

### ISSUE THREAD

**Issue by maunzCache** [terraform-aws-modules/terraform-aws-lambda#475]
## Is your request related to a new offering from AWS?
Is this functionality available in the [AWS provider for Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)? See [CHANGELOG.md](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md), too.
  - No 🛑: please wait to file a request until the functionality is available in the AWS provider

## Is your request related to a problem? Please describe.
I am used to set up a python project with the `pyproject.toml` format which was introduced by [PEP518](https://peps.python.org/pep-0518/). As far as i understood, this should be a common interface to bundle a python project, maybe deprecating the known `setup.py` solution in a near future. With the introduction of this file, a `requirements.txt` file which is commonly used for managing a projects dependencies is no longer required, as the `pyproject.toml` capture requirements and optional dependencies.

The current package solution `package.py` is - from what i understand - only proven to work with pip dependencies managed by a `requirements.txt`.

## Describe the solution you'd like.
Basically, i'd like to have support to create packages based on a `pyproject.toml` file.

## Describe alternatives you've considered.
Replace my common approach with the convention of using a `requirements.txt` file.

## Additional context
I am fully aware that this is a custom solution for this module and you may not want to support every possible variante of packaging code. My rational to create this request is that a `pyproject.toml` file seems to be the proper way to manage a python project, while the `requirements.txt` is rather a convention not backed up by a PEP. So i'd love to see a project adapting to that PEP.

**Comment by maunzCache** [comment]
Please note that it might not be required to write any code for this but write down an example at least.
